src/h5.h

```diff
@@ -15,46 +17,12 @@ using std::array;
 using std::vector;
 using std::string;
```

Comment:
I realise it is not in these changes, but now is a good time to fix it. Could these `using` statements go in the cxx file to avoid polluting the h5 namespace? Only `unique_ptr` and `string` are used in the header and not in many places.